### ISSUE TITLE
Use in tsx with vue3, props type error

### ISSUE THREAD

**Issue by init-center** [iconify/iconify#95]
![image](https://user-images.githubusercontent.com/21151034/142395056-d9be81ad-0127-44d5-9c9d-a4c4208cbcda.png)
The icon can be displayed on the page, but the type is wrong, there is no "icon" property on the props

**Comment by cyberalien** [comment]
Sorry for slow response.

I'm not sure how to implement it. Component accepts all attributes. If attribute applies to icon, such as `icon` attribute, it is used to generate icon. Other attributes are passed to `svg` element.

**Comment by chenzhq** [comment]
same error

**Comment by Ttou** [comment]
add `./types/iconify.d.ts` 

```ts
interface _Icon {
  new (): {
    $props: import('@iconify/vue/dist/iconify').IconProps
  }
}

declare module '@iconify/vue' {
  export const Icon: _Icon

 // export you want
  export { addCollection } from '@iconify/vue/dist/iconify'
}
```

**Comment by tachibana-shin** [comment]
> Sorry for slow response.
> 
> I'm not sure how to implement it. Component accepts all attributes. If attribute applies to icon, such as `icon` attribute, it is used to generate icon. Other attributes are passed to `svg` element.

This problem has become more serious in the latest `volar` version when Volar.js started dropping support for undefined attributes I believe the `vue-tsc` package will be updated soon as well making vue 3 does not accept components.

Can you fix it or do you want me to fix it with a pull request?

**Comment by cyberalien** [comment]
I tried code above, but it didn't solve issue for me. If that fix works for you, I can add it or you can make a pr. If not, better solutions are welcome.

**Comment by cyberalien** [comment]
Thanks for pr!

Published version `4.1.1-dev` (install `@iconify/vue@dev`) with your fix.

Doesn't seem to change anything for me. Can you check if it fixes issue for you?

**Comment by tachibana-shin** [comment]
> Thanks for pr!
> 
> Published version `4.1.1-dev` (install `@iconify/vue@dev`) with your fix.
> 
> Doesn't seem to change anything for me. Can you check if it fixes issue for you?

version `4.1.1-dev` works perfectly for me. Maybe your lps server language hasn't been reloaded:
![image](https://user-images.githubusercontent.com/45375496/229325777-e9aa49c4-1973-45da-8e16-261b6ec50d64.png)
![image](https://user-images.githubusercontent.com/45375496/229325783-313bf179-d7a4-4da2-99e8-bec25cc79905.png)
![image](https://user-images.githubusercontent.com/45375496/229325786-99cf2d65-a7ae-4ffa-a4d6-43886ba4dea8.png)

**Comment by cyberalien** [comment]
Thanks a lot!

Merged, published `4.1.1`.
